File: TestSample/java/src/main/java/helloworld/speakers/JaSpeaker.java
package helloworld.speakers;

import helloworld.Person;

public final class JaSpeaker extends Person {
    public JaSpeaker(String name) {
        super(name, "ja");
    }

    @Override
    public String speak() {
        return "こんにちは、世界！";
    }
}

File: TestSample/rust/src/speakers/ja_speaker.rs
use crate::person::{Person, PersonBase};

pub struct JaSpeaker {
    base: PersonBase,
}

impl JaSpeaker {
    pub fn new(name: &str) -> Self {
        Self { base: PersonBase::new(name, "ja") }
    }
}

impl Person for JaSpeaker {
    fn name(&self) -> &str { &self.base.name }
    fn language(&self) -> &str { &self.base.language }
    fn speak(&self) -> String { String::from("こんにちは、世界！") }
}
